### DISCUSSION TITLE
Using Tailwind programmatically

### DISCUSSION THREAD

**Discussion by andresberrios** [tailwindlabs/tailwindcss#5241] [Help]
Hello! Thanks for the great work on Tailwind!

Recently for a project we found ourselves needing to generate Tailwind CSS on demand for a list of specific classes which we are adding dynamically onto the page. After tinkering around with different options, we ended up making this:

https://github.com/andresberrios/tailwindcss-lib

It works pretty well and we're loving what we can do with it (and loving the JIT engine!), but it's a pretty hacky implementation and the resulting file size is not as small as it could be.

I was wondering if there might be some easier way to accomplish this that we missed, or if maybe some functionality like this could be provided by Tailwind directly in a simpler and cleaner way.

Basically this is about providing direct access to the Tailwind CSS generation engine, and decoupling it from node.js APIs like `fs` so it can be used in the browser too. Trimming down on dependencies not essential for this purpose would be a plus, and it would be ok to limit the functionality if necessary, like not supporting the `aot` engine, removing the need for purging, etc.

Any thoughts? 😁
